pymeasure/instruments/agilent/agilent4156.py

```diff
@@ -0,0 +1,1052 @@
+
+        :param trace_list: A list of channel variables whose measured data should be saved. A maximum of 2 variables are allowed. If only one variable is being saved, a string can be specified.
+
+        .. code-block:: python
+
+            instr.save_var(['VA', 'VB'])
+        """
+        self.write(":PAGE:DISP:MODE LIST")
+        if isinstance(trace_list, list):
+            try:
+                len(trace_list) > 2
+            except:
+                raise RuntimeError('Maximum of 2 variables allowed')
+            else:
+                for name in trace_list:
+                    self.write(":PAGE:DISP:DVAR \'{}\'".format(name))
+        elif isinstance(trace_list, str):
+            self.write(":PAGE:DISP:DVAR \'{}\'".format(trace_list))
+        else:
+            raise TypeError(
+                'Must be a string if only one variable is saved, or else a list if'
+                'multiple variables are being saved.'
+            )
+
+    @property
+    def data_variables(self):
+        """
+        Gets a string list of data variables for which measured data
+        is available. This looks for all the variables saved by the :meth:`~.Agilent4156.save` and :meth:`~.Agilent4156.save_var` methods and returns it. This is useful for creation of dataframe headers.
+
+        :returns: List
+
+        .. code-block:: python
+
+            header = instr.data_variables
+        """
+        dlist = self.ask(":PAGE:DISP:LIST?").split(',')
+        dvar = self.ask(":PAGE:DISP:DVAR?").split(',')
+        varlist = dlist + dvar
+        return list(filter(None, varlist))
+
+    def get_data(self, path=None):
+        """
+        Gets the measurement data from the instrument after completion. If the measurement period is set to :code:`INF` in the :meth:`~.Agilent4156.measure` method, then the measurement must be stopped using :meth:`~.Agilent4156.stop` before getting valid data.
+
+        :param path: Path for optional data export to CSV.
+        :returns: Pandas Dataframe
+
+        .. code-block:: python
+
+            df = instr.get_data(path='./datafolder/data1.csv')
+        """
+        if int(self.ask('*OPC?')):
+            header = self.data_variables
+        self.write(":FORM:DATA ASC")
+        # recursively get data for each variable
+        for i, listvar in enumerate(header):
+            data = self.values(":DATA? \'{}\'".format(listvar))
+            time.sleep(0.01)
+            if i == 0:
+                lastdata = data
+            else:
+                data = np.column_stack((lastdata, data))
+                lastdata = data
+
+        df = pd.DataFrame(data=data, columns=header, index=None)
+        if path is not None:
+            _, ext = os.path.splitext(path)
+            if ext != ".csv":
+                path = path + ".csv"
+            df.to_csv(path, index=False)
+
+        return df
+
+##########
+# CHANNELS
+##########
+
+
+class smu(Instrument):
```

Comment:
By convention, classes should start with an uppercase letter -- this makes it easier to identify when something is a class. I would suggest naming this `SMU`.